tdrs-backend/tdpservice/parsers/schema_defs/tribal_tanf/t2.py

```diff
@@ -0,0 +1,740 @@
+from ...fields import TransformField, Field
+from ...row_schema import RowSchema
+from ... import validators
+from tdpservice.search_indexes.models.tribal import Tribal_TANF_T2
+
+
+t2 = SchemaManager(
+    schemas=[
+        RowSchema(
+            model=Tribal_TANF_T2,
+            preparsing_validators=[
+                validators.hasLength(122),
+            ],
+            postparsing_validators=[
+                validators.validate__FAM_AFF__SSN(),
+                validators.if_then_validator(
+                    condition_field="FAMILY_AFFILIATION",
+                    condition_function=validators.matches(1),
+                    result_field="SSN",
+                    result_function=validators.validateSSN(),
+                ),
+                validators.if_then_validator(
+                    condition_field="FAMILY_AFFILIATION",
+                    condition_function=validators.isInLimits(1, 3),
+                    result_field="RACE_HISPANIC",
+                    result_function=validators.isInLimits(1, 2),
+                ),
+                validators.if_then_validator(
+                    condition_field="FAMILY_AFFILIATION",
+                    condition_function=validators.isInLimits(1, 3),
+                    result_field="RACE_AMER_INDIAN",
+                    result_function=validators.isInLimits(1, 2),
+                ),
+                validators.if_then_validator(
+                    condition_field="FAMILY_AFFILIATION",
+                    condition_function=validators.isInLimits(1, 3),
+                    result_field="RACE_ASIAN",
+                    result_function=validators.isInLimits(1, 2),
+                ),
+                validators.if_then_validator(
+                    condition_field="FAMILY_AFFILIATION",
+                    condition_function=validators.isInLimits(1, 3),
+                    result_field="RACE_BLACK",
+                    result_function=validators.isInLimits(1, 2),
+                ),
+                validators.if_then_validator(
+                    condition_field="FAMILY_AFFILIATION",
+                    condition_function=validators.isInLimits(1, 3),
+                    result_field="RACE_HAWAIIAN",
+                    result_function=validators.isInLimits(1, 2),
+                ),
+                validators.if_then_validator(
+                    condition_field="FAMILY_AFFILIATION",
+                    condition_function=validators.isInLimits(1, 3),
+                    result_field="RACE_WHITE",
+                    result_function=validators.isInLimits(1, 2),
+                ),
+                validators.if_then_validator(
+                    condition_field="FAMILY_AFFILIATION",
+                    condition_function=validators.isInLimits(1, 3),
+                    result_field="MARITAL_STATUS",
+                    result_function=validators.isInLimits(1, 5),
+                ),
+                validators.if_then_validator(
```

Comment:
noticing here that when this kind of error pops up that parent_minor_child data element is missing in the name columns.  not sure yet if this applies to all the cat 3 scenarios like this
```
error_message:
if family affiliation :1 validator1 passed then PARENT_MINOR_CHILD None is not larger or equal to 1 and smaller or equal to 3.

Item Name:
,family affiliation

Internal Variable Name:
,FAMILY_AFFILIATION

```

Reply:
note to self and data team: this appears to be a common error among tribal section 1 files.

Reply:
@elipe17 did you see this one?

Reply:
@ADPennington I have a fix for this. But, I am going to put it into the Tribal S4 PR because it is going to introduce a new migration. To keep the dependencies clean and avoid any painful migration issues, putting it in S4 makes the most sense.

Reply:
ok will test this in #2762